metadata-ingestion/src/datahub/ingestion/source/sql/trino.py

```diff
@@ -64,20 +65,41 @@
 KNOWN_CONNECTOR_PLATFORM_MAPPING = {
     "clickhouse": "clickhouse",
     "hive": "hive",
-    "postgresql": "postgres",
-    "mysql": "mysql",
+    "glue": "glue",
     "iceberg": "iceberg",
+    "mysql": "mysql",
+    "postgresql": "postgres",
     "redshift": "redshift",
     "bigquery": "bigquery",
+    "snowflake_distributed": "snowflake",
+    "snowflake_parallel": "snowflake",
+    "snowflake_jdbc": "snowflake",
 }
 
-TWO_TIER_CONNECTORS = ["clickhouse", "hive", "mysql", "iceberg"]
+TWO_TIER_CONNECTORS = ["clickhouse", "hive", "glue", "mysql", "iceberg"]
+
+PROPERTIES_TABLE_SUPPORTED_CONNECTORS = ["hive", "iceberg"]
 
 # Type JSON was introduced in trino sqlalchemy dialect in version 0.317.0
 if version.parse(trino.__version__) >= version.parse("0.317.0"):
     register_custom_type(datatype.JSON, RecordTypeClass)
 
 
+@functools.lru_cache()
+def get_catalog_connector_name(
+    connection: Connection, catalog_name: str
+) -> Optional[str]:
+    query = dedent(
+        """
+        SELECT *
+        FROM "system"."metadata"."catalogs"
+    """
+    ).strip()
+    res = connection.execute(sql.text(query))
+    catalog_connector_dict = {row.catalog_name: row.connector_name for row in res}
```

Comment:
split this up into two methods - one to generate catalog_connector_dict, and the second doing the lookup

the lru_cache annotation should be on the first one